parser/src/parser/grammar.lalrpop

```diff
@@ -256,6 +257,42 @@ EvaluatorSegmentBindings: (SourceSpan, Vec<Span<(Identifier, usize)>>) = {
     <l:@L> "[" "]" <r:@R> => (span!(l, r), vec![]),
 }
 
+// FUNCTIONS
+// ================================================================================================
+
+Function: Function = {
+    <l:@L> "fn" <name: FunctionIdentifier> "(" <params: FunctionBindings> ")" "->" <ty: FunctionBindingType> "{" <body: FunctionBody> "}" <r:@R>
+        => Function::new(span!(l, r), name, params, ty, body)
+}
+
+FunctionBindings: Vec<(Identifier, Type)> = {
+    <l:@L> <params: Comma<FunctionBinding>> <r:@R> => params,
+}
+
+FunctionBinding: (Identifier, Type) = {
+    <name: Identifier> ":" <ty: FunctionBindingType> => (name, ty),
+}
+
+FunctionBindingType: Type = {
+    <l:@L> "felt" <r:@R> => Type::Felt,
+    <l:@L> "felt" <size: Size> <r:@R> => Type::Vector(size as usize),
+    <l:@L> "felt" "[" <row_size: Num_u64> "," <col_size: Num_u64> "]" <r:@R> => Type::Matrix(row_size as usize, col_size as usize),
+}
+
+FunctionBody: Vec<Statement> = {
+    // TODO: validate
+    <l:@L> <stmts: StatementBlock> <r:@R> =>? {
```

Comment:
Question: what does this `TODO` mean here?

Reply:
I think it's actually already done, I just forgot to remove the TODO, I'll do that and then merge this.